addon/ng2/blueprints/component/index.js

```diff
@@ -16,22 +19,59 @@ module.exports = {
   },
 
   locals: function (options) {
+    //TODO: pull value from config
+    this.styleExt = 'css';
+    
     return {
       dynamicPath: this.dynamicPath.dir.replace(this.dynamicPath.appRoot, ''),
-      flat: options.flat
+      flat: options.flat,
+      styleExt: this.styleExt,
+      isLazyRoute: !!options.isLazyRoute,
+      isAppComponent: !!options.isAppComponent
     };
   },
+  
+  files: function() {
+    var fileList = getFiles.call(this);
+    
+    if (this.options.flat) {
+      fileList = fileList.filter(p => p.indexOf('index.ts') <= 0);
+    }
 
+    return fileList;
+  },
+  
   fileMapTokens: function (options) {
     // Return custom template variables here.
     return {
       __path__: () => {
         var dir = this.dynamicPath.dir;
         if (!options.locals.flat) {
           dir += path.sep + options.dasherizedModuleName;
+          
+          if (options.locals.isLazyRoute) {
+            var dirParts = dir.split(path.sep);
+            dirParts[dirParts.length - 1] = `+${dirParts[dirParts.length - 1]}`;
+            dir = dirParts.join(path.sep);
+          }
         }
+        this.appDir = dir.replace(`src${path.sep}client${path.sep}`, '');
         return dir;
+      },
+      __styleext__: () => {
+        return options.locals.styleExt;
       }
     };
+  },
+  
+  afterInstall: function(options) {
+    if (!options.flat) {
+      var filePath = path.join('src', 'client', 'system-config.ts');
+      return this.insertIntoFile(
+        filePath,
+        `  '${this.appDir}',`,
+        { before: '  //barrel registration' }
```

Comment:
So I had some issues with `insertIntoFile` which prompted me to create my version for the route injection. For this here I'd rather not use a token comment but see if you can do the following logic instead:

```typescript
      var filePath = path.join('src', 'client', 'system-config.ts');
      // Read the file
      var content = fs.readFileSync(filePath, 'utf-8');

      // Find the barrel definition, including all its current values, and replace the last line
      // with the registration, including a `,` if none is there.
      const regex = new RegExp(''
          + '('  // Keep the current declaration intact.
            + /(?:const|let|var)\s+/.source  // Declaration keyword.
            + /barrels(?::\s*\S*)?\s*=\s*/.source  // Variable name, followed by `=`.
            + /\[\s*\n[\s\S]*?/.source  // The current declaration, up to the closest `];`
          + ')'
          + /,?\s*\n\s*\];/.source  // The last optional ',' followed by `];` itself.
          , 'm');  // Multi-line of course.
      content = content.replace(regex, (m0, m1) => {
        return m1 + `,\n  '${this.appDir}'\n];`
      });

      fs.writeFileSync(filePath, content, 'utf-8');
```

What do you think?